lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -14,12 +16,366 @@ import 'semantics.dart';
+    }
+    _resizeQueue!.add((
+      representation: this,
+      targetSize: size,
+    ));
+  }
+
+  @override
+  void cleanUp() {
+    _domText.remove();
+  }
+
+  static List<_QueuedSizeUpdate>? _resizeQueue;
+
+  static void _updateSizes() {
+    final List<_QueuedSizeUpdate>? queue = _resizeQueue;
+
+    // Eagerly reset the queue before doing any work. This ensures that if there
+    // is an unexpected error while processing the queue, we don't end up in a
+    // cycle that grows the queue indefinitely. Worst case, some text nodes end
+    // up incorrectly sized, but that's a smaller problem compared to running
+    // out of memory.
+    _resizeQueue = null;
+
+    assert(
+      queue != null && queue.isNotEmpty,
+      '_updateSizes was called with an empty _resizeQueue. This should never '
+      'happend. If it does, please file an issue at '
+      'https://github.com/flutter/flutter/issues/new/choose',
+    );
+
+    if (queue == null || queue.isEmpty) {
+      // This should not happen, but if it does (e.g. something else fails and
+      // caused the post-update callback to be called with an empty queue), do
+      // not crash.
+      return;
+    }
+
+    final List<_Measurement> measurements = <_Measurement>[];
+
+    // Step 1: measure all spans in a single batch prior to updating their CSS
+    //         styles. This way, all measurements are taken with a single reflow
+    //         Interleaving measurements with updates, will cause the browser to
+    //         reflow the page between measurements.
+    for (final _QueuedSizeUpdate update in queue) {
+      // Both clientWidth/Height and offsetWidth/Height provide a good
+      // approximation for the purposes of sizing the focus ring of the text,
+      // since there's no borders or scrollbars. The `offset` variant was chosen
+      // mostly because it rounds the value to `int`, so the value is less
+      // volatile and therefore would need fewer updates.
+      //
+      // getBoundingClientRect() was considered and rejected, because it provides
+      // the rect in screen coordinates but this scale adjustment needs to be
+      // local.
+      final double domWidth = update.representation._domText.offsetWidth;
+      final double domHeight = update.representation._domText.offsetHeight;
+      measurements.add((
+        representation: update.representation,
+        domSize: ui.Size(domWidth, domHeight),
+        targetSize: update.targetSize,
+      ));
+    }
+
+    // Step 2: update all spans at a batch without taking any further DOM
+    //         measurements, which avoids additional reflows.
+    for (final _Measurement measurement in measurements) {
+      final SizedSpanRepresentation representation = measurement.representation;
+      final double domWidth = measurement.domSize.width;
+      final double domHeight = measurement.domSize.height;
+      final ui.Size targetSize = measurement.targetSize;
```

Comment:
Can't. `measurement.domSize.width` doesn't destructure.

Reply:
Oops! It can still be done like this:
```suggestion
      final _Measurement(
        :SizedSpanRepresentation representation,
        domSize: ui.Size(width: double domWidth, height: double domHeight),
        :ui.Size targetSize,
      ) = measurement;
```
but I think it's less readable than what you had originally.